### ISSUE TITLE
feat: collaborative mode

### ISSUE THREAD

**Issue by h0tw4t3r** [transmissions11/flux#38]
**Is your feature request related to a problem? Please describe.**
It would be useful to collaborate with someone on the same page, since **flux** is getting more structurization features with #8 and possibly #18.
**Describe the solution you'd like**
A collaborative mode can be implemented with a button view, which on click generates a unique link and opens a non-closable modal on the corner of a screen with a link that the user can copy to send to anyone to collaborate on the prompts.

Something similar to what **codesandbox** has:
<img width="197" alt="Screenshot 2023-04-02 at 13 42 58" src="https://user-images.githubusercontent.com/35642018/229348048-0e7056b5-b361-4f01-80c1-250aeb5b9869.png">

**Describe alternatives you've considered**

**Additional context**
- https://reactflow.dev/docs/examples/interaction/collaborative/

**Comment by mabry1985** [comment]
I agree! I've been considering putting together a socket.io server to accomplish this. I think it would be cool if each user could bring their own API key and have the possibility for working on separate conversations at the same time

**Comment by bmorphism** [comment]
could automerge / another CRDT implementation maybe?
